idf_build_apps/app.py

```diff
@@ -292,6 +308,26 @@ def size_json_path(self) -> t.Optional[str]:
     def config(self) -> t.Optional[str]:
         return self.config_name
 
+    def _process_override_sdkconfig_file(self):
+        self.override_sdkconfig = App.override_sdkconfig.copy()
+
+        for f in App.override_sdkconfig_file:
+            if not os.path.isabs(f):
+                f = os.path.join(self.work_dir, f)
+            if not os.path.isfile(f):
+                continue
+            with open(f) as fr:
+                for line in fr:
+                    m = self.SDKCONFIG_LINE_REGEX.match(line)
+                    if not m:
+                        continue
+                    if (
+                        m.group(1) in self.override_sdkconfig
+                        and App.override_priority == OverridePriority.OVERRIDE_SDKCONFIG
+                    ):
+                        continue
+                    self.override_sdkconfig[m.group(1)] = m.group(2)
```

Comment:
The statements here is not very intuitive.

how about

```python
if App.override_priority == OverridePriority.OVERRIDE_SDKCONFIG:
    base_dict = self.override_sdkconfig.copy()
    override_dict = self.override_sdkconfig_file_items.copy()

else:
    base_dict = self.override_sdkconfig_file_items.copy()
    override_dict = self.override_sdkconfig.copy()
```

and then do a merge dict `override_dict --override--> base_dict`?

Reply:
Agree, there are some unintuitive things here.
@hfudev, what do you think about this way to do it?
```python
    def _process_override_sdkconfig_file(self):
        self.override_sdkconfig = App.override_sdkconfig.copy()
        override_sdkconfig_file_items = {}

        for f in App.override_sdkconfig_file:
            if not os.path.isabs(f):
                f = os.path.join(self.work_dir, f)
            if not os.path.isfile(f):
                continue
            with open(f) as fr:
                for line in fr:
                    m = self.SDKCONFIG_LINE_REGEX.match(line)
                    if not m:
                        continue
                    override_sdkconfig_file_items[m.group(1)] = m.group(2)

        if App.override_priority == OverridePriority.OVERRIDE_SDKCONFIG:
            self.override_sdkconfig = override_sdkconfig_file_items | self.override_sdkconfig 
        else:
            self.override_sdkconfig = self.override_sdkconfig | override_sdkconfig_file_items 
````